api/types/events/events.proto

```diff
@@ -1779,3 +1793,92 @@ message SessionUpload {
     // URL is where the url the session event data upload is at
     string SessionURL = 5 [ (gogoproto.jsontag) = "url" ];
 }
+
+// Identity matches github.com/gravitational/teleport/lib/tlsca.Identity except
+// for RouteToApp and RouteToDatabase which are nullable and Traits which is
+// represented as a google.protobuf.Struct (still containing a map from string
+// to strings).
+message Identity {
+    // Username is a username or name of the node connection
+    string Username = 1 [ (gogoproto.jsontag) = "user,omitempty" ];
+    // Impersonator is a username of a user impersonating this user
+    string Impersonator = 2 [ (gogoproto.jsontag) = "impersonator,omitempty" ];
+    // Groups is a list of groups (Teleport roles) encoded in the identity
+    repeated string Groups = 3 [ (gogoproto.jsontag) = "roles,omitempty" ];
+    // Usage is a list of usage restrictions encoded in the identity
+    repeated string Usage = 4 [ (gogoproto.jsontag) = "usage,omitempty" ];
+    // Principals is a list of Unix logins allowed.
+    repeated string Principals = 5 [ (gogoproto.jsontag) = "logins,omitempty" ];
+    // KubernetesGroups is a list of Kubernetes groups allowed
+    repeated string KubernetesGroups = 6 [ (gogoproto.jsontag) = "kubernetes_groups,omitempty" ];
+    // KubernetesUsers is a list of Kubernetes users allowed
+    repeated string KubernetesUsers = 7 [ (gogoproto.jsontag) = "kubernetes_users,omitempty" ];
+    // Expires specifies whenever the session will expire
+    google.protobuf.Timestamp Expires = 8 [
+        (gogoproto.stdtime) = true,
+        (gogoproto.nullable) = false,
+        (gogoproto.jsontag) = "expires"
+    ];
+    // RouteToCluster specifies the target cluster
+    // if present in the session
+    string RouteToCluster = 9 [ (gogoproto.jsontag) = "route_to_cluster,omitempty" ];
+    // KubernetesCluster specifies the target kubernetes cluster for TLS
+    // identities. This can be empty on older Teleport clients.
+    string KubernetesCluster = 10 [ (gogoproto.jsontag) = "kubernetes_cluster,omitempty" ];
+    // Traits hold claim data used to populate a role at runtime.
+    google.protobuf.Struct Traits = 11
```

Comment:
Could we use a `map<string, string>` instead? `Struct` is best avoided, as there are no format guarantees once it is in play.

Reply:
It's not a `map[string]string`, it's a `map[string][]string` - we could use `LabelValues` from `api/types/wrappers/wrappers.proto` but `events.proto` hasn't imported anything so far, and we're already using `google.protobuf.Struct` for `UserLogin.IdentityAttributes` and `AccessRequestCreate.Annotations` (both are `map[string][]string` just like `Traits`).

What do you think?

Reply:
`wrappers.LabelValues` was already used with gogoproto's custom type mapping for another `Traits` (and other assorted `Labels` fields), so I added the dependency onto `wrappers.proto`.